File: clients/java/signalr/src/test/java/com/microsoft/signalr/TestUtils.java
// Copyright (c) .NET Foundation. All rights reserved.
// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.

package com.microsoft.signalr;

class TestUtils {
    static HubConnection createHubConnection(String url) {
        return createHubConnection(url, new MockTransport(), new NullLogger(), true, new TestHttpClient());
    }

    static HubConnection createHubConnection(String url, Transport transport) {
        return createHubConnection(url, transport, new NullLogger(), true, new TestHttpClient());
    }

    static HubConnection createHubConnection(String url, Transport transport, Logger logger, boolean skipNegotiate, HttpClient client) {
        HttpHubConnectionBuilder builder = HubConnectionBuilder.create(url)
                .withTransport(transport)
                .withHttpClient(client)
                .shouldSkipNegotiate(skipNegotiate)
                .withLogger(logger);

        return builder.build();
    }
}

File: src/SignalR/clients/cpp/test/signalrclient-e2e-tests/test_utils.cpp
// Copyright (c) .NET Foundation. All rights reserved.
// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.

#pragma once

#include "stdafx.h"
#include <string>
#include "cpprest/details/basic_types.h"
#include "cpprest/asyncrt_utils.h"
#include <chrono>

utility::string_t url;

void get_url(int argc, utility::char_t* argv[])
{
    url = U("http://localhost:42524/");

    for (int i = 0; i < argc; ++i)
    {
        utility::string_t str = argv[i];

        auto pos = str.find(U("url="));
        if (pos != std::string::npos)
        {
            url = str.substr(pos + 4);
            break;
        }
    }
}
